### ISSUE TITLE
[Bug]: propElementValues option of react/jsx-curly-brace-presence doesn't catch JSX fragments as prop values

### ISSUE THREAD

**Issue by mfigard** [yannickcr/eslint-plugin-react#3951]
### Is there an existing issue for this?

- [x] I have searched the existing issues and my issue is unique
- [x] My issue appears in the command-line and not only in the text editor

### Bug description

The `propElementValues: 'always'` option in the `react/jsx-curly-brace-presence` rule does not flag JSX fragments (`<>...</>`) passed as prop values without curly braces.

### Example code

```tsx
import * as React from 'react';

const AttentionIcon = () => <span>⚠️</span>;

export const TestComponent = () => {
    return (
        <div>
            {/* This JSX fragment should be flagged but isn't */}
            <div
                title=<>
                    <AttentionIcon /> Installation not completed
                </>
            />

            {/* This is correctly ignored (already has braces) */}
            <div title={<>Fragment with braces</>} />

            {/* This regular JSX element is correctly flagged */}
            <div title=<div>Regular element without braces</div> />

        </div>
    );
};
```
### Error encountered

ESLint runs without flagging any errors for the JSX fragments used as prop values without curly braces.

### Reproducing case

The error should be visible in an IDE but isn't. No error is reported when running `eslint`:

```bash
# With this ESLint configuration:
# "react/jsx-curly-brace-presence": ["error", { "propElementValues": "always" }]

eslint TestComponent.tsx
# Result: No errors reported (should report one error)
```

### Expected behavior description

When using the `propElementValues: 'always'` configuration, ESLint should flag the JSX fragments as prop values without curly braces and offer auto-fix suggestions to wrap the JSX elements in curly braces.

### Example

```tsx
export const TestComponent = () => {
    return (
        <div>
            {/* Should be auto-fixed to have braces */}
            <div
                title={
                    <>
                        <AttentionIcon /> Installation not completed
                    </>
                }
            />
        </div>
    );
};
```

The rule should report one error and offer an auto-fix suggestion. The error and auto-fix should be available in any IDE.

### eslint-plugin-react version

v7.37.5

### eslint version

v8.57.0

### node version

v22.13.1

**Comment by mfigard** [comment]
👋 I tried implementing a fix here: https://github.com/jsx-eslint/eslint-plugin-react/pull/3952 but the tests are failing because the JSX fragment is not parsed properly ([job here](https://github.com/jsx-eslint/eslint-plugin-react/actions/runs/17796329623/job/50585164963?pr=3952)):

```
  1) jsx-curly-brace-presence
       invalid
         <App horror=<><div /></> />
// features: [no-ts], parser: default, options: [{"props":"always","children":"always","propElementValues":"always"}]:

      AssertionError: A fatal parsing error occurred: Parsing error: Unexpected token >
      + expected - actual

      -false
      +true
      
      at testInvalidTemplate (node_modules/eslint/lib/testers/rule-tester.js:477:28)
      at Context.<anonymous> (node_modules/eslint/lib/testers/rule-tester.js:557:25)
```

Not sure what the best way forward is as I'd rather not change the repo's test configuration. Let me know if there's anything I can do to help!
